runtime.go

```diff
@@ -87,15 +91,25 @@ type Runtime interface {
 	// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#name-section%E2%91%A0
 	CompileModule(ctx context.Context, binary []byte) (CompiledModule, error)
 
-	// InstantiateModule instantiates the module or errs if the configuration was invalid.
+	// InstantiateModule instantiates the module or errs for reasons including
+	// exit or validation.
 	//
 	// Here's an example:
-	//	mod, _ := n.InstantiateModule(ctx, compiled, wazero.NewModuleConfig().WithName("prod"))
+	//	mod, _ := n.InstantiateModule(ctx, compiled, wazero.NewModuleConfig().
+	//		WithName("prod"))
+	//
+	// # Errors
 	//
-	// While CompiledModule is pre-validated, there are a few situations which can cause an error:
+	// While CompiledModule is pre-validated, there are a few situations which
+	// can cause an error:
 	//   - The module name is already in use.
-	//   - The module has a table element initializer that resolves to an index outside the Table minimum size.
+	//   - The module has a table element initializer that resolves to an index
+	//     outside the Table minimum size.
 	//   - The module has a start function, and it failed to execute.
+	//   - The module was compiled to WASI and exited. Even on success, exit
```

Comment:
Do you think it would be useful to not error in that case? 

It seems like the error would always be non-nil for WASI programs, or depend on compiler specific behavior (e.g calling proc_exit(0) after main returns).

It reminds me of how the net/http.Server always returns a non nil error when calling ListenAndServe... it's such a pain to deal with.

Reply:
I was looking through the history and didn't see a reason why. I think at first it was trying to be like a CancelationError. SGTM to remove that

Reply:
I think it is less impact to do this as a special case, like for start functions invoked inside Instantiate, not for all functions.

Reply:
Agreed